code/core/src/highlight/useHighlights.ts

```diff
@@ -63,9 +63,11 @@ export const useHighlights = ({
   rootId?: string;
   storybookRootId?: string;
 }) => {
-  // Clean up any existing instance of useHighlights
+  if (globalThis.__STORYBOOK_HIGHLIGHT_INITIALIZED) {
+    return;
+  }
 
-  (globalThis as any).__STORYBOOK_HIGHLIGHT_TEARDOWN?.();
+  globalThis.__STORYBOOK_HIGHLIGHT_INITIALIZED = true;
```

Comment:
From what you told me, this is only needed because of our stories testing this functionality, correct?

I'm wondering if this could be avoided somehow, or at least have a comment explaining what this global variable is for.

Reply:
I think this happens either way. In our own Storybook I've seen preview.js and runtime.js both execute `useHighlights`, and they also do it twice. That might be a bug in Storybook or be specific to our own Storybook, but either way I feel safer knowing we can only ever have one instance. Of course refactoring to a decorator might be the real answer here.